lib/modules/pipeline/index.js

```diff
@@ -29,16 +29,35 @@ class Pipeline {
       'get',
       '/embark-api/file',
       (req, res) => {
-        if (!fs.existsSync(req.query.path) || !req.query.path.startsWith(fs.dappPath())) {
-          return res.send({error: 'Path is invalid'});
+        try {
+          this.apiGuardBadFile(req.query.path);
+        } catch (error) {
+          return res.send({error: error.message});
         }
+        
         const name = path.basename(req.query.path);
         const content = fs.readFileSync(req.query.path, 'utf8');
         res.send({name, content, path: req.query.path});
 
       }
     );
 
+    plugin.registerAPICall(
+      'post',
+      '/embark-api/folders',
```

Comment:
I think it should be `'/embark-api/folder'` (without the `s`) as we create one folder

Reply:
I remember, you did the same comment on `POST /embark-api/files`
REST assume it is plural when you do a POST

Reply:
Yeah, ok, I'm just a grumpy man then :)